commitrmnocb/types.go

```diff
@@ -0,0 +1,270 @@
+
+	// A map from chain selectors to each chain's consensus OffRamp next sequence number
+	OffRampNextSeqNums map[cciptypes.ChainSelector]cciptypes.SeqNum
+
+	// A map from chain selectors to each chain's consensus f (failure tolerance)
+	FChain map[cciptypes.ChainSelector]int
+}
+
+// GasPricesArray returns a list of gas prices
+func (co ConsensusObservation) GasPricesArray() []cciptypes.GasPriceChain {
+	gasPrices := make([]cciptypes.GasPriceChain, 0, len(co.GasPrices))
+	for chain, gasPrice := range co.GasPrices {
+		gasPrices = append(gasPrices, cciptypes.NewGasPriceChain(gasPrice.Int, chain))
+	}
+
+	return gasPrices
+}
+
+// TokenPricesArray returns a list of token prices
+func (co ConsensusObservation) TokenPricesArray() []cciptypes.TokenPrice {
+	tokenPrices := make([]cciptypes.TokenPrice, 0, len(co.TokenPrices))
+	for tokenID, tokenPrice := range co.TokenPrices {
+		tokenPrices = append(tokenPrices, cciptypes.NewTokenPrice(tokenID, tokenPrice.Int))
+	}
+
+	return tokenPrices
+}
+
+type OutcomeType int
+
+const (
+	ReportIntervalsSelected OutcomeType = iota + 1
+	ReportGenerated
+	ReportEmpty
+	ReportInFlight
+	ReportTransmitted
+	ReportTransmissionFailed
+)
+
+type Outcome struct {
+	OutcomeType                     OutcomeType                 `json:"outcomeType"`
+	RangesSelectedForReport         []ChainRange                `json:"rangesSelectedForReport"`
+	RootsToReport                   []cciptypes.MerkleRootChain `json:"rootsToReport"`
+	OffRampNextSeqNums              []plugintypes.SeqNumChain   `json:"offRampNextSeqNums"`
+	TokenPrices                     []cciptypes.TokenPrice      `json:"tokenPrices"`
+	GasPrices                       []cciptypes.GasPriceChain   `json:"gasPrices"`
+	ReportTransmissionCheckAttempts uint                        `json:"reportTransmissionCheckAttempts"`
+}
+
+// Sort all fields of the given Outcome
+func (o Outcome) sort() {
+	sort.Slice(o.RangesSelectedForReport, func(i, j int) bool {
+		return o.RangesSelectedForReport[i].ChainSel < o.RangesSelectedForReport[j].ChainSel
+	})
+	sort.Slice(o.RootsToReport, func(i, j int) bool {
+		return o.RootsToReport[i].ChainSel < o.RootsToReport[j].ChainSel
+	})
+	sort.Slice(o.OffRampNextSeqNums, func(i, j int) bool {
+		return o.OffRampNextSeqNums[i].ChainSel < o.OffRampNextSeqNums[j].ChainSel
+	})
+	sort.Slice(o.TokenPrices, func(i, j int) bool {
+		return o.TokenPrices[i].TokenID < o.TokenPrices[j].TokenID
+	})
+	sort.Slice(o.GasPrices, func(i, j int) bool {
+		return o.GasPrices[i].ChainSel < o.GasPrices[j].ChainSel
+	})
+}
+
+// Encode encodes an Outcome deterministically
+func (o Outcome) Encode() ([]byte, error) {
+
+	// Sort all lists to ensure deterministic serialization
+	o.sort()
+
+	encodedOutcome, err := json.Marshal(o)
+	if err != nil {
+		return nil, fmt.Errorf("failed to encode Outcome: %w", err)
+	}
+
+	return encodedOutcome, nil
+}
+
+func DecodeCommitPluginOutcome(b []byte) (Outcome, error) {
```

Comment:
nit: Hopefully get rid of everything that says `CommitPlugin`
```suggestion
func DecodeOutcome(b []byte) (Outcome, error) {
```